lib/api.es6.js

```diff
@@ -50,6 +47,102 @@ const login = (apiOptions, username, pass) => {
   });
 };
 
+const register = (apiOptions, username, password, email, newsletter) => {
+  return new Promise((r, x) => {
+    if (!apiOptions.oauthAppOrigin) {
+      x('Please set up a Reddit Oauth App, and pass in its URL as oauthAppOrigin to config.');
+    }
+
+    if (!apiOptions.clientId) {
+      x('Please set up a Reddit Oauth App, and pass in its id as clientId to config.');
+    }
+
+    if (!apiOptions.clientSecret) {
+      x('Please set up a Reddit Oauth App, and pass in its secret as clientSecret to config.');
+    }
+
+    const data = {
+      user: username,
+      passwd: password,
+      passwd2: password,
+      api_type: 'json',
+    };
+
+    if (email) {
+      data.email = email;
+    }
+
+    if (newsletter) {
+      data.newsletter_subscribe = true;
+
+      if (!email) {
+        x('NEWSLETTER_NO_EMAIL');
+      }
+    }
+
+    const endpoint = `${apiOptions.origin}/api/register`;
+    const s = btoa(`${apiOptions.clientId}:${apiOptions.clientSecret}`);
+    const basicAuth = `Basic ${s}`;
+    const headers = {
+      'User-Agent': apiOptions.userAgent,
+      'Authorization': basicAuth,
+      ...apiOptions.defaultHeaders,
+    };
+
+    try {
+      const timeout = 20000;
+      superagent
+        .post(endpoint)
+        .set(headers)
+        .type('form')
+        .send(data)
+        .timeout(10000)
+        .end((err, res) => {
```

Comment:
i don't think there is a `catch`, just a `then`. for what it's worth, i actually prefer using `.end` with superagent so that it doesn't make it look like superagent uses real promises.